scripts/pipelines/github/templates/build/build-pipeline.cfg

```
@@ -0,0 +1,34 @@
+# Mandatory flags.
+mandatoryFlags="$pipelineName,$localDirectory,$language,"
+# Path to the templates.
+templatesPath="scripts/pipelines/github/templates/build"
+# YAML file name.
+yamlFile="build-pipeline.yml"
+# Script name.
+scriptFile="build.sh"
+# Source branch.
+sourceBranch="feature/build-pipeline"
+
+# Function that copies the script to build the application.
+function copyScript {
+    # Copy the script.
+    cp "${hangarPath}/${templatesPath}/${language}-${scriptFile}" "${localDirectory}/${scriptFilePath}/${scriptFile}"
+}
+
+# Function that adds the variables to be used in the pipeline.
+function addPipelineVariables {
+    # if the user did not specify a custom target-directory 
+    # we default to the language specific defaults
+
+    if test -z $targetDirectory
+    then
+        case $language in 
+            node | angular) targetDirectory="./" ;;
+            quarkus*) targetDirectory="./target/" ;;
+            *) echo -e "${red}Error: Specified language '${language}' is not supported." >&2; exit 1
+        esac
+    fi
+
+    # We cannot use a variable in the definition of resource in the pipeline so we have to use a placeholder to replace it with the value we need
+    envsubst '${pipelineName} ${targetDirectory}' < "${hangarPath}/${templatesPath}/${yamlFile}.template" > "${localDirectory}/${pipelinePath}/${yamlFile}"
```

This won't work correctly because we cannot do two envsubst this will just override the file already copied.

But with @jbellver99 we talked about it and I did some changes to the pipeline generator so that this function is executed before the copy yaml file so you just need to export the variable that you want to change.

Also, we changed the way we us the envsubst in the copy yaml file, we use two list, one for the general variables that need to be change and one with the spcific var
```suggestion
    specificEnvSubstList='${targetDirectory}'
```

---

Done